pkg/validate/validate.go

```diff
@@ -184,9 +184,10 @@ func validateAgentPoolProfile(app *api.AgentPoolProfile) (errs []error) {
 		errs = append(errs, fmt.Errorf("invalid properties.agentPoolProfiles[%q].role %q", app.Name, app.Role))
 	}
 
+	// must match generate.go limit geneartion code
```

Comment:
I was looking for something like `if _, found := DefaultVMSizeKubeArguments[app.VMSize]; !found { // error }` here

In which case you need to change the meaning of DefaultVMSizeKubeArguments a little - it's more like the map of valid VM names and various associated parameters for each.

I don't know if pkg/api is the right location for that map.

Reply:
If not in `pkg/api`. I see only 2 places, either `pkg/config` or `pkg/api`. And we should agree where we keep all defaults in one place. So we would not end-up calling some defaults from `api` as const values, and some as variables from `config`.